File: src/main/java/com/ascargon/rocketshow/midi/ActionTriggerMidiSongSelect.java
package com.ascargon.rocketshow.midi;

import lombok.Getter;
import lombok.Setter;

/**
 * Triggered by a MIDI song select message (0xF3), which masters use to announce the song they are
 * about to play. Song select carries no channel, so the channel of the trigger is ignored.
 */
@Getter
@Setter
public class ActionTriggerMidiSongSelect extends ActionTriggerMidi {

    // If null -> all songs
    private Integer song;

}

File: src/main/webapp/src/app/models/action-trigger-midi-song-select.ts
import { ActionTriggerMidi } from "./action-trigger-midi";

export class ActionTriggerMidiSongSelect extends ActionTriggerMidi {
  song: number;

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    this.song = data.song;

    if (!this.song) {
      this.song = 0;
    }
  }

  toJSON() {
    return { actionTriggerMidiSongSelect: { ...this } };
  }
}
